### DISCUSSION TITLE
Shearing or other multmatrix equivalent

### DISCUSSION THREAD

**Discussion by abejfehr** [jscad/OpenJSCAD.org#1139] [Q&A] [answered]
Hello, I'd like to translate this OpenSCAD project into JSCAD: https://github.com/wbu/drawer_organizer

I've tried using the [openscad-translator](https://github.com/jscad/openscad-openjscad-translator), but since the OpenSCAD code contains assignments I've resorted to doing it by hand instead.

The original makes use of `multmatrix` in a few places (like [this](https://github.com/wbu/drawer_organizer/blob/master/drawer_organizer.scad#L153)). It looks like it does a shear and a translate conditionally.

Is there any way to shear in JSCAD or am I out of luck?

**Comment by hrgdavor** [comment, marked as answer]
Quickly thinking about it, it should be possible.
https://openjscad.xyz/docs/module-modeling_transforms.html#.transform

transforms allows you to supply your own transform matrix and shear should work fine.

notable difference is that in openscad it is two dimensional 4x4 array, and in jscad a simple 16 length array

```js
const jscad = require('@jscad/modeling')
const {transform, translate} = jscad.transforms
const {cube} = jscad.primitives


const main = ()=>{
  const skew = 0.1
  const matrix = [
        1,skew,0,-skew,
        0,1,0,0,
        0,0,1,0,
        0,0,0,1 
        ]
  return [
     transform(matrix,cube({size:10})),
     translate([-20,0,0],cube({size:10})),
     ]
}

module.exports = {main}
```

![image](https://user-images.githubusercontent.com/2480762/187086504-7629637e-dbcc-44db-9825-6611fc7d5bdb.png)

**Reply by abejfehr** [reply]
Thanks for the quick reply, this works well!
